StripeConnect/StripeConnect/Source/Internal/Webview/ConnectComponentWebView.swift

```diff
@@ -46,18 +47,31 @@ class ConnectComponentWebView: ConnectWebView {
         config.allowsInlineMediaPlayback = true
 
         super.init(frame: .zero, configuration: config)
+        componentManager.registerForAppearanceUpdates(self)
         guard let publishableKey = componentManager.apiClient.publishableKey else {
             assertionFailure("A publishable key is required. For more info, see https://stripe.com/docs/keys")
             return
         }
         addMessageHandlers()
         addNotificationObservers()
-        load(.init(url: StripeConnectConstants.connectJSURL(component: componentType.rawValue, publishableKey: publishableKey)))
+        if loadContent {
+            load(.init(url: StripeConnectConstants.connectJSURL(component: componentType.rawValue, publishableKey: publishableKey)))
+        }
     }
 
+    func updateAppearance(appearance: Appearance) {
+        var appearance = appearance
+        appearance.traitCollection = self.traitCollection
+        sendMessage(UpdateConnectInstanceSender.init(payload: .init(locale: webLocale.webIdentifier, appearance: appearance)))
+    }
+    
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        updateAppearance(appearance: componentManager.appearance)
```

Comment:
🚫 Blocker
I think that this needs to get wrapped in `DispatchQueue.main.async { }` – as of iOS 14, I believe that this is called off the main thread when the app is foregrounded, which causes a crash since it's accessing UIKit classes off the main thread. It's possible this may have been fixed in later iOS versions, but I suspect we still need to handle it since we're supporting 15+